crates/re_space_view/src/data_query_blueprint.rs

```diff
@@ -0,0 +1,401 @@
+use nohash_hasher::IntSet;
+use re_data_store::{EntityProperties, EntityPropertyMap, EntityTree};
+use re_log_types::{EntityPath, EntityPathExpr};
+use re_viewer_context::{
+    DataResult, EntitiesPerSystem, EntitiesPerSystemPerClass, SpaceViewClassName,
+};
+use slotmap::SlotMap;
+use smallvec::SmallVec;
+
+use crate::{
+    blueprint::QueryExpressions, DataQuery, DataResultHandle, DataResultNode, DataResultTree,
+    PropertyResolver,
+};
+
+/// An implementation of [`DataQuery`] that is built from a collection of [`QueryExpressions`]
+///
+/// During execution it will walk an [`EntityTree`] and return a [`DataResultTree`]
+/// containing any entities that match the input list of [`EntityPathExpr`]s.
+///
+/// Any exact expressions are included in the [`DataResultTree`] regardless of whether or not they are found
+/// this allows UI to show a "no data" message for entities that are being explicitly looked for but
+/// not found.
+///
+/// The results of recursive expressions are only included if they are found within the [`EntityTree`]
+/// and for which there is a valid `ViewPart` system. This keeps recursive expressions from incorrectly
+/// picking up irrelevant data within the tree.
+pub struct DataQueryBlueprint {
+    pub blueprint_path: EntityPath,
+    pub space_view_class_name: SpaceViewClassName,
+    pub expressions: QueryExpressions,
+}
+
+impl DataQueryBlueprint {
+    pub const OVERRIDES_PREFIX: &str = "overrides";
+}
+
+impl DataQuery for DataQueryBlueprint {
+    fn execute_query(
+        &self,
+        property_resolver: &impl PropertyResolver,
+        ctx: &re_viewer_context::StoreContext<'_>,
+        entities_per_system_per_class: &EntitiesPerSystemPerClass,
+    ) -> DataResultTree {
+        re_tracing::profile_function!();
+
+        let mut data_results = SlotMap::<DataResultHandle, DataResultNode>::default();
+
+        let query_override = property_resolver.resolve_root_override(ctx);
+        let entity_overrides = property_resolver.resolve_entity_overrides(ctx);
+
+        if let Some(per_system_entity_list) =
+            entities_per_system_per_class.get(&self.space_view_class_name)
+        {
```

Comment:
Good catch! I hadn't looked closely enough at `identify_entities_per_system_per_class` to realize that we don't get a list if there are no entities that match.

I'll switch this to use a default empty system map in that case so we still add exact-matches in those cases.

Also this matches the current auto-behavior but as we build out a fancier query spec and/or view-system overrides this will probably want to evolve.